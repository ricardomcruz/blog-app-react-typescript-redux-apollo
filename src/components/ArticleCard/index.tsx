import classNamesBind from 'classnames/bind';
import { sanitize } from 'dompurify';
import parse from 'html-react-parser';
import Card from 'react-bootstrap/Card';
import { PostWithComments } from '../../_modules/posts/models/Post';
import styles from './styles.module.scss';

const cx = classNamesBind.bind(styles);

export interface ArticleCardProps {
  post: PostWithComments
  darkMode: boolean
}

const ArticleCard = (
  {
    post: {
      title,
      publish_date,
      content,
      author,
    },
    darkMode
  }: ArticleCardProps
) => {

  return (
    <Card className={cx('p-4', 'mb-4', { "bg-dark": darkMode })}>
      <h2>
        {title}
      </h2>
      <h5 className='mt-2 text-secondary'>{author}</h5>
      <div>
        <div>{parse(sanitize(content))}</div>
        <div className={cx('mt-4')}>
          <small>{publish_date}</small>
        </div>
      </div>
    </Card>
  );
}

export default ArticleCard;
