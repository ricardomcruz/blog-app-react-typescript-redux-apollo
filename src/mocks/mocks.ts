
import { Comment, CommentDTO } from '../../_modules/comments/models/Comment';
import { Post } from '../../_modules/posts/models/Post';

export const postMock: Post = {
  id: 1,
  title: 'title',
  author: 'author',
  publish_date: 'publish_date',
  description: 'description',
  content: 'content'
}

export const commentMockDTO: CommentDTO = {
  postId: 2,
  user: 'user',
  date: 'date',
  content: 'content',
};

export const commentMock: Comment = {
  ...commentMockDTO,
  id: 1,
  parent_id: 2,
};

export const postWithCommentsMock: Post = {
 ...postMock,
 comments: commentMock
};