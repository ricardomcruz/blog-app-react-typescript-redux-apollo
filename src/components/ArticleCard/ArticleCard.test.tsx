/** * @jest-environment jsdom */

import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { postWithCommentsMock } from '../../mocks/mocks';
import ArticleCard from './index';

describe('ArticleCard', () => {

  it('should render the Article Card', () => {
    const { asFragment }= render(
      <ArticleCard
        darkMode={false}
        post={postWithCommentsMock}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
