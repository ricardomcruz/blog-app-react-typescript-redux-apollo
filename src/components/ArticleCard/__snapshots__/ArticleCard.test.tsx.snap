// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard should render the Article Card 1`] = `
<DocumentFragment>
  <div
    class="p-4 mb-4 card"
  >
    <h2>
      title
    </h2>
    <h5
      class="mt-2 text-secondary"
    >
      author
    </h5>
    <div>
      <div>
        content
      </div>
      <div
        class="mt-4"
      >
        <small>
          publish_date
        </small>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
